src/v2/cookbook/debugging-in-vscode.md

```diff
@@ -0,0 +1,132 @@
+
+Install and create a project with the [vue-cli](https://github.com/vuejs/vue-cli) by running
+
+```
+npm install -g @vue/cli
+# or
+yarn global add @vue/cli
+
+vue create my-project
+```
+
+Change to the newly created application directory and open VS Code.
+
+```
+cd my-project
+
+# if you have an alias set up (recommended)
+code .
+```
+
+### Update your webpack configuration
+
+Before you can debug your Vue components from VS Code you need to update the generated webpack config to build sourcemaps that contains more information for our debugger.
+
+* Go to `config/index.js` and find the `devtool` property. Update it to:
+
+```json
+devtool: 'source-map',
+```
+
+### Configure launch.json File
+
+Click on the Debugging icon in the Activity Bar to bring up the Debug view, then click on the gear icon to configure a launch.json file, selecting **Chrome** for the environment. Replace content of the generated launch.json with the following two configurations:
+
+![Add Chrome Configuration](/images/config_add.png)
+
+```json
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "type": "chrome",
+      "request": "launch",
+      "name": "vuejs: chrome",
+      "url": "http://localhost:8080",
+      "webRoot": "${workspaceFolder}/src",
+      "breakOnLoad": true,
+      "sourceMapPathOverrides": {
+        "webpack:///src/*": "${webRoot}/*"
+      }
+    }
+  ]
+}
+```
+
+## Start Debugging An Application
+
+1. Set a breakpoint in **src/components/HelloWorld.vue** on `line 90` where the `data` function returns a string.
+
+![Breakpoint Renderer](/images/breakpoint_set.png)
+
+2. Open your favorite terminal at the root folder and serve the app using Vue CLI:
+
+```
+npm start
+```
+
+3. Go to the Debug view, select the **'vuejs: chrome'** configuration, then press F5 or click the green play button.
+
+4. Your breakpoint should now be hit as the new instance of Chrome opens `http://localhost:8080`.
+
+![Breakpoint Hit](/images/breakpoint_hit.png)
+
+## Alternative Patterns
+
+### Vue Devtools
+
+There are other methods of debugging, varying in complexity. The most popular and simple of which is to use the excellent [vue-devtools](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd). Some of the benefits of working with the devtools are that they enable you to live-edit data properties and see the changes reflected immediately. The other major benefit is the ability to do time travel debugging for Vuex.
+
+![Devtools Timetravel Debugger](/images/devtools-timetravel.gif)
+
+<p class="tip">There are a couple of gotchas to note for Vue Devtools: 
+- If the page uses a production/minified build of Vue.js, devtools inspection is disabled by default so the Vue pane won't show up. 
+- To make it work for pages opened via file:// protocol, you need to check "Allow access to file URLs" for this extension in Chrome's extension management panel</p>
```

Comment:
Whenever there's another doc that explains something in more detail, it's best to link to it so that we don't have to maintain multiple versions of the same instructions. In this case, I'd just link to the Vue Devtools README, since it contains all this information already.

Reply:
Good idea.

Reply:
Actually, I'm thinking about keeping a note about it not working for minified version, but maybe changing the wording to say "for instance, using a CDN" because I get this question _a ton_ and it really wouldn't hurt to have it in one more place, IMO.